final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import requests
+import argparse
+import json
+import logging
+import pycodestyle
+from bs4 import BeautifulSoup
+
+
+log = logging.getLogger("Log")
+
+
+def find_news(url):
+    """Parses news"""
+    logging.info("Searching for news on " + url)
+    try:
+        resp = requests.get(url)
+        soup = BeautifulSoup(resp.content, features="xml")
+        items = soup.findAll('item')
+        log.info(str(len(items)) + " pieces of news have been found")
+    except requests.exceptions.MissingSchema:
+        log.error("Wrong URL")
+        print("Wrong URL: " + url)
+        sys.exit()
+    except requests.exceptions.ConnectionError:
+        log.error("Can't connect to the " + url)
+        print("Can't connect to the " + url)
+        sys.exit()
+    return items
+
+
+def find_channel(url):
+    resp = requests.get(url)
+    soup = BeautifulSoup(resp.content, features="xml")
+    channel = soup.find('channel')
+    return channel
+
+
+def collect_news(items, limit) -> list:
+    """Sorts the components of the news into dictionaries and adds them into a list"""
+    news = []
+    for item in items:
+        try:
+            news_item = {}
+            news_item['title'] = item.title.text
+            news_item['date'] = item.pubDate.text
+            soup = BeautifulSoup(item.description.text,'lxml')
+            news_item['description'] = soup.text
+            news_item['link'] = item.link.text
+            if item.thumbnail:
+                news_item['image'] = item.thumbnail['url']
+            elif item.content:
+                news_item['image'] = item.content['url']
+            else:
+                news_item['image'] = item.enclosure['url']
+            if soup.find('img'):
+                news_item['alt'] = soup.find('img')['alt']
+            elif item.find('media:text'):
+                soup = BeautifulSoup(item.find('media:text').text, 'lxml')
+                news_item['alt'] = soup.find('img')['alt']
+            else:
+                news_item['alt'] = None
+        except TypeError:
+            log.error("Could not load an image")
+            news_item['image'] = "No image"
+        except AttributeError:
+            log.error("Probably no description")
+            news_item['description'] = "No description"
+        news.append(news_item)
+        if len(news) == limit:
+            break
+    return news
+
+
+def print_news(list_of_news, channel):
+    print("Feed: " + channel.title.text + '\n\n')
+    for item in list_of_news:
+        print("Title: " + item['title'])
+        print("Date: " + item['date'])
+        print("Description: " + item['description'])
+        print("Image: " + item['image'])
+        if item['alt']:
+            print("Alt: " + item['alt'])
+        print("Link: " + item['link'])
+        print('------------------')
+
+
+def json_convert(news):
+    """Converts the news into JSON format"""
+    with open("JSON.json", "w") as write_file:
+        json.dump(news, write_file)
```

Comment:
`--json` option shiould just print parsed news to console, but in json format. No need actually write it to file.